internal/deployer/builder.go

```diff
@@ -0,0 +1,109 @@
+// Package deployer for project deployment
+package deployer
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/grid3-go/workloads"
+	"github.com/threefoldtech/grid_proxy_server/pkg/client"
+	"github.com/threefoldtech/grid_proxy_server/pkg/types"
+	"github.com/threefoldtech/zos/pkg/gridtypes"
+	"github.com/threefoldtech/zos/pkg/gridtypes/zos"
+)
+
+var (
+	vmFlist      = "https://hub.grid.tf/aelawady.3bot/abdulrahmanelawady-gridify-test-latest.flist"
+	vmCPU        = 2
+	vmMemory     = 2 * 1024
+	vmRootfsSize = 5 * 1024
+	vmEntrypoint = "/init.sh"
+	vmPublicIP   = true
+	vmPlanetary  = true
+)
+
+func constructNodeFilter() types.NodeFilter {
+	nodeStatus := "up"
+	freeMRU := uint64(2)
+	freeHRU := uint64(5)
+	freeIPs := uint64(1)
+	domain := true
+
+	filter := types.NodeFilter{
+		FarmIDs: []uint64{1},
+		Status:  &nodeStatus,
+		FreeMRU: &freeMRU,
+		FreeHRU: &freeHRU,
+		FreeIPs: &freeIPs,
+		Domain:  &domain,
+	}
+	return filter
+}
+
+func findNode(gridProxyClient client.Client) (uint32, error) {
+	filter := constructNodeFilter()
+	nodes, _, err := gridProxyClient.Nodes(filter, types.Limit{})
+	if err != nil {
+		return 0, err
+	}
+	if len(nodes) == 0 {
+		return 0, errors.New("no node with free resources available")
+	}
+
+	node := uint32(nodes[0].NodeID)
+	return node, nil
+}
+
+func constructNetwork(projectName string, node uint32) workloads.ZNet {
```

Comment:
can be 
type Builder struct { 
    projectName 
    node ..
    ....

}

then https://refactoring.guru/design-patterns/builder/go/example